src/vitamin.cpp

```diff
@@ -47,6 +47,56 @@ int vitamin::severity( int qty ) const
     return 0;
 }
 
+std::string vitamin::get_string_level( int qty ) const
+{
+    float norm_level = 1.0f;
+    if( type_ == vitamin_type::VITAMIN ) {
+        if( id_ == "vitA" ) {
+            norm_level = 2.0f;
+        } else if( id_ == "vitB" ) {
+            norm_level = 2.75f;
+        } else if( id_ == "vitC" ) {
+            norm_level = 2.2f;
+        } else if( id_ == "iron" ) {
+            norm_level = 1.23f;
+        } else if( id_ == "calcium" ) {
+            norm_level = 1.1f;
+        }
+        const float v_level = norm_level * qty / disease_[0].first;
+        if( v_level != 0 ) {
+            std::string message = string_format( _( "%s level is %.2f times %s than normal." ), name(),
+                                                 std::abs( v_level ), ( v_level < 0  ? _( "higher" ) : _( "lower" ) ) );
+            const int sev = severity( qty );
+            if( sev ) {
+                const std::vector<std::string> s = { "", _( "Minor" ), _( "Severe" ), _( "Extreme" ) };
+                message += string_format( " %s %s.", s[std::abs( sev )],
+                                          ( sev < 0 ? _( "deficiency" ) : _( "excess" ) ) );
+            }
+            return message;
+        } else {
+            return string_format( _( "%s level is normal." ), name() );
+        }
+    }
+    if( type_ == vitamin_type::TOXIN ) {
+        if( id_ == "mutant_toxin" ) {
+            if( qty > ( disease_excess_[2].first ) ) {
+                return _( "Deadly levels of unknown toxins detected." );
+            } else if( qty > ( disease_excess_[1].first ) ) {
+                return _( "Extreme levels of unknown toxins detected." );
+            } else if( qty > ( disease_excess_[0].first ) ) {
+                return _( "Dangerous levels of unknown toxins detected." );
+            } else if( qty > ( disease_excess_[0].first / 2 ) ) {
+                return _( "Significant levels of unknown toxins detected." );
+            } else if( qty > ( disease_excess_[0].first / 4 ) ) {
+                return _( "Traces of unknown toxins detected." );
+            } else {
+                return "";
+            }
+        }
+    }
+    return string_format( "%s level is %d.", name(), qty );
```

Comment:
This appears to just give you the exact vitamin level if the vitamin is either a `disease_excess` vitamin (addressed earlier), or quantity is 0.
I don't think the vitamin quantity should ever be directly displayed, because it's not information that makes much sense to the player, and they have no need to know it - I'd suggest changing this message to `%s level is normal`, or something similar.

Reply:
Just changed it to output empty string. Unknown vitamins shouldn't be shown by Analyzer at all.